### PR TITLE
Add Read-Only Property to UtilityMenuItems

### PR DESCRIPTION
**Pull request by amichaelparker**
### Summary
Adds a read-only property to the menu item shape, allowing menu items to be clicked on and copy/pasted without closing the menu on click.

Closes [#359](https://github.com/cerner/terra-framework/issues/359)

### CODE REVIEW

File: packages/terra-application-utility/src/utility/_UtilityMenuItem.jsx [hunk 1]
```diff
@@ -127,6 +132,7 @@ class UtilityMenuItem extends React.Component {
     const bodyItemClassNames = cx([
       { 'header-utility-body-item': variant === Utils.VARIANTS.HEADER },
       { 'menu-utility-body-item': variant === Utils.VARIANTS.MENU },
+      { 'read-only': isReadOnly },
```
Comment (emilyrohrbough) [inline packages/terra-application-utility/src/utility/_UtilityMenuItem.jsx]: If this is readOnly, would we need to change the role type of the list item and remove the click / keydown handlers?

Comment (amichaelparker) [reply packages/terra-application-utility/src/utility/_UtilityMenuItem.jsx]: Thinking something like this to address this:
```
const roles = isReadOnly ? 'note' : 'button';
/* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/no-noninteractive-element-to-interactive-role */
const renderBodyItem = (fill, wrapOnKeyDown, handleSelection) => (
  <li
    {...customProps}
    tabIndex={!isReadOnly ? '0' : undefined}
    key={itemKey}
    onClick={!isReadOnly ? event => handleSelection(event, itemKey) : undefined}
    onKeyDown={!isReadOnly ? wrapOnKeyDown(itemKey, onKeyDown) : undefined}
    role={roles}
    className={bodyItemClassNames}
    aria-label={title}
    ref={this.setItemNode}
  >
    <Arrange
      fitStart={leftInset ? <IconCheckmark className={checkmarkClassNames} /> : null}
      fill={fill}
      fillAttributes={{ className: cx('menu-item-fill') }}
      fitEnd={rightInset ? <IconChevronRight className={chevronClassNames} /> : null}
      align="center"
      className={arrangeClassNames}
    />
  </li>);
```
Lint pops up an error about the tabindex on a non-interactive element, but if I do what is in the code above, that creates an accessibility issue (screen readers won't be able to tab to it to read it). Otherwise, I can just disable the rule for that line.

File: packages/terra-application-utility/src/utility/_UtilityMenuItem.module.scss [hunk 2]
```diff
@@ -145,4 +145,17 @@
   .menu-item-fill {
     width: 100%;
   }
+
+  .read-only {
+    cursor: text;
+
+    &:hover {
+      background-color: var(--terra-application-header-utility-menu-body-item-hover-background-color);
```
Comment (yuderekyu) [inline packages/terra-application-utility/src/utility/_UtilityMenuItem.module.scss]: we should sync with UX, but I think a read only item will never have hover and focus states. We'd probably set `background-color: transparent` and `box-shadow: none`, instead of opening them up for themes

Comment (amichaelparker) [reply packages/terra-application-utility/src/utility/_UtilityMenuItem.module.scss]: I thought so too, but wasn't sure if I should leave the option there just in case. Will take it out!

Comment (neilpfeiffer) [reply packages/terra-application-utility/src/utility/_UtilityMenuItem.module.scss]: I need to review PR further, but generally anything that is "read-only" or non-interactive elements on the page never should have have any visual state changes (pointer/color,background-color/shadow etc.) for hover, focus, active, etc. (for all themes)

File: packages/terra-application-utility/src/utility/_UtilityMenuItem.jsx [hunk 3]
```diff
@@ -149,15 +155,17 @@ class UtilityMenuItem extends React.Component {
...
       <li
         {...customProps}
         tabIndex="0"
         key={itemKey}
-        onClick={event => handleSelection(event, itemKey)}
-        onKeyDown={wrapOnKeyDown(itemKey, onKeyDown)}
-        role="button"
+        onClick={!isReadOnly ? event => handleSelection(event, itemKey) : undefined}
+        onKeyDown={!isReadOnly ? wrapOnKeyDown(itemKey, onKeyDown) : undefined}
+        role={roles}
```
Comment (yuderekyu) [inline packages/terra-application-utility/src/utility/_UtilityMenuItem.jsx]: Nitpick - since we're doing inline conditionals, it would be fine do the same for role
```jsx
role={isReadOnly ? 'note' : 'button'}
```

I don't have much accessibility knowledge, but we may rework how these roles are set. Looks like we we may open a first class prop to configure this: https://github.com/cerner/terra-framework/issues/272. That is out of scope though, just something to ponder

File: packages/terra-application-utility/src/utility/_UtilityMenuItem.jsx (hunk 1 above)
Comment (bjankord) [reply packages/terra-application-utility/src/utility/_UtilityMenuItem.jsx]: What you have looks good to me. role="note" seems fine to me as well, hadn't heard of it before so I had to look it up.

File: packages/terra-application-utility/src/utility/_UtilityMenuItem.jsx (hunk 3 above)
Comment (bjankord) [reply packages/terra-application-utility/src/utility/_UtilityMenuItem.jsx]: Inlining this conditional should be fine. Let's tackle opening the role prop as a first-class prop in a separate PR.

File: packages/terra-application-utility/src/utility/_UtilityMenuItem.module.scss (hunk 2 above)
Comment (bjankord) [reply packages/terra-application-utility/src/utility/_UtilityMenuItem.module.scss]: All the focus styles have been removed, the only remaining question I have is should we still allow people to focus to it via keyboard tab? I'm thinking we drop that functionality from read-only items in the menu as it should be non-interactive, e.g. no clicking on it to trigger something.

Comment (amichaelparker) [reply packages/terra-application-utility/src/utility/_UtilityMenuItem.module.scss]: I was originally thinking altering the tabindex would also hide the information from a screen reader, but I realized you can't tab to it anyway, so I threw the conditional back in there.

**Comment by yuderekyu** [comment]
Let's document `isReadOnly` in the readme. https://github.com/cerner/terra-framework/blob/master/packages/terra-application-utility/docs/README.md

**Comment by amichaelparker** [comment]
Added a (probably rough draft) explanation of the isReadOnly prop.

**Comment by emilyrohrbough** [comment]
Looks good. Kinda random, but I am noticing the themed screenshots have grown in height? Any clue why?

**Comment by amichaelparker** [comment]
> Looks good. Kinda random, but I am noticing the themed screenshots have grown in height? Any clue why?

This was some odd behavior I came across while working on this - I've only tested this with the examples, but essentially if you add items to the scroll list, and the number of items makes the scroll list longer than the containing menu height, it actually increases the height of the menu as well.

In the comparison screenshots, the read-only menu item has been added to the new pics, so it has pushed the total height of the menu down as well. If anyone has any thoughts on addressing that, I can slip in a fix here, or it can be another PR.

**Comment by mmalaker** [comment]
Functional verification completed. The utility menu example displays and functions as expected with the read-only item included.
